contracts/xasset/src/error.rs

```diff
@@ -75,10 +75,19 @@ pub enum Error {
 
     // Payment exceeds interest due
     PaymentExceedsInterestDue = 24,
-    
+
     // "Interest must be paid before debt can be repaid"
     InterestMustBePaidFirst = 25,
 
     // "Insufficient XLM to pay interest"
     InsufficientXLMForInterest = 26,
+
+    // Approval needed for interest repayment
+    InsufficientApprovedXLMForInterestRepayment = 27,
+
+    // Invoking XLM SAC contract failed
+    XLMHostInvocationFailed = 28,
+
+    // Interest repaid must be greater than 0
```

Comment:
If these are doc comments, do they show up various places? They would be useful messages to expose, and we should probably make them doc comments even if they don't yet show up anywhere, to at least communicate that they are user-facing comments.
```suggestion
    /// Interest repaid must be greater than 0
```

Reply:
Looks like it only passes the error enum name to the message section of the error. Would be cool if it passed the comment. In index.js: ```export const Errors = {
    1: { message: "InsufficientCollateralization" },
    2: { message: "CDPAlreadyExists" },```